VDA-StorageNode-Registry/contracts/StorageNodeRegistry.sol

```diff
@@ -0,0 +1,1098 @@
+     */
+    function updateNodeIssueFee(uint value) external payable onlyOwner {
+        if (value == 0 || value == _slotInfo.NODE_ISSUE_FEE) {
+            revert InvalidValue();
+        }
+        uint orgFee = _slotInfo.NODE_ISSUE_FEE;
+        _slotInfo.NODE_ISSUE_FEE = value;
+
+        emit UpdateNodeIssueFee(orgFee, value);
+    }
+
+    /**
+     * @dev see { IStorageNodeRegistry }
+     */
+    function getTotalIssueFee() external view returns(uint) {
+        return _slotInfo.totalIssueFee;
+    }
+
+    /**
+     * @dev see { IStorageNodeRegistry }
+     */
+    function withdrawIssueFee(address to, uint amount) external payable onlyOwner {
+        if (amount > _slotInfo.totalIssueFee) {
+            revert InvalidValue();
+        }
+        IERC20Upgradeable(vdaTokenAddress).transfer(to, amount);
+
+        _slotInfo.totalIssueFee = _slotInfo.totalIssueFee - amount;
+
+        emit WithdrawIssueFee(to, amount);
+    }
+
+    function getSameNodeLogDuration() external view returns(uint) {
+        return _slotInfo.SAME_NODE_LOG_DURATION;
+    }
+
+    /**
+     * @dev see { IStorageNodeRegistry }
+     */
+    function updateSameNodeLogDuration(uint value) external payable onlyOwner {
+        if (value == 0 || value == _slotInfo.SAME_NODE_LOG_DURATION) {
+            revert InvalidValue();
+        }
+        uint orgVal = _slotInfo.SAME_NODE_LOG_DURATION;
+        _slotInfo.SAME_NODE_LOG_DURATION = value;
+
+        emit UpdateSameNodeLogDuration(orgVal, value);
+    }
+
+    /**
+     * @dev see { IStorageNodeRegistry }
+     */
+    function logNodeIssue(
+        address didAddress,
+        address nodeAddress,
+        uint reasonCode,
+        bytes calldata requestSignature,
+        bytes calldata requestProof
+    ) external {
+        {
+            // Check whether nodeDID is registered
+            uint nodeId = _didNodeId[nodeAddress];
+            if (nodeId == 0) {
+                revert InvalidDIDAddress();
+            }
+
+            bytes memory params = abi.encodePacked(didAddress, nodeAddress, reasonCode);
+            verifyRequest(didAddress, params, requestSignature, requestProof);
+        }
+
+        
+        DIDLogInformation storage logs = _didLogs[didAddress];
+        // Check whether there are 4 logs in 24 horus
+        if (logs._issueList.length >= 4) {
+            uint earlistTime = logs._issueList[logs.index].time;
+            if (block.timestamp - earlistTime < 24 hours) {
+                revert TimeNotElapsed();
+            }
+        }
+        // Check 1 hour condition for same node
+        for (uint i; i < logs._issueList.length;) {
+            if (logs._issueList[i].nodeDID == nodeAddress && 
+                (block.timestamp - logs._issueList[i].time) < _slotInfo.SAME_NODE_LOG_DURATION) {
+                revert InvalidSameNodeTime();
+            }
+            unchecked { ++i; }
+        }
+
+        // Add or update
+        if (logs._issueList.length < 4) {
```

Comment:
Can you explain this code? Why 4?

Reply:
Here the `4` is the available number of logs in the 24 hours.
In that code `logs._issueList` keeps the logs in 24 hours. At first it is an empty array.
So if there are logs less than 4, it pushs the logs in the array.
After the length of array reached 4, it replaced the oldest log with the new one.

Reply:
Ok, can you please add a comment that explains that. Make `4` a variable that owner can change as well.

Reply:
Already updated as a variable.